CONTRIBUTING.md

```diff
@@ -151,30 +136,72 @@ Once you've run this, your repository should be set for most changes (i.e., you
 
 ### 4. Make the changes you've discussed
 
-Try to keep the changes focused to the issue. We've found that working on a [new branch][link_branches] for each issue makes it easier to keep your changes targeted. 
+Try to keep the changes focused to the issue.  
+We've found that working on a [new branch][link_branches] for each issue makes it easier to keep your changes targeted.
+Using a new branch allows you to follow the standard "fork/branch/commit/pull-request/merge" GitHub 
+workflow when making changes.
+[This guide][link_gitworkflow] provides a useful overview for this workflow.
+Before making a new branch, make sure your master is up to date.
 
-Using a new branch allows you to follow the standard "fork/branch/commit/pull-request/merge" GitHub workflow when making changes. [This guide][link_gitworkflow] provides a useful overview for this workflow.
+```
+git checkout master
+git fetch upstream master
+git merge upstream/master
+```
+
+Then, make your new branch.
+
+```
+git checkout -b MYBRANCH
+```
+
+As you're making changes, make sure your branch is kept up to date with 
+
+```
+git fetch upstream master
+git merge upstream/master
+```
+
+If you know what rebasing is, please only use it for changes that haven't been pushed.
+If you don't know what rebasing is, don't do it at all, as it is the easiest way to make your 
+repository a disaster zone.
+Please make sure to review the `tedana` [style conventions](#style-guide) and test your changes.
+
+If you are new to ``git``, there are several GUI git clients that you may find helpful, such as
+- [GitKraken][link_git_kraken]
+- [GitHub Desktop][link_github_desktop]
+- [SourceTree][link_source_tree]
 
-Before creating your pull request, please make sure to review the `tedana` [style conventions](#style-guide).
 
 ### 5. Test your changes
 
-#### Changes to code
+You can run style checks by running the following:
+```
+flake8 $TEDANADIR/tedana
+```
 
-For changes to the codebase, we suggest using our development Docker container which will run all the necessary checks and tests to ensure your code is ready to be merged into `tedana`!
-(This does require that you have a local install of [Docker](https://www.docker.com/products/docker-desktop).)
-You can run all the checks with:
+and unit/integration tests by running `pytest` (more details below).
+If you know a file will test your change, you can run only that test.
+Alternatively, running all unit tests is relatively quick and should be
+fairly comprehensive.
+Running all `pytest` tests will be useful for pre-pushing checks.
+When you open a Pull Request, CircleCI will run all tests.
 
 ```
-docker run --tty --rm -v ${PWD}:/tedana tedana/tedana-dev:latest run_all_tests
+# All tests; final checks before pushing
```

Comment:
I'm not sure I follow, sorry, could you elaborate?

Reply:
I think (in line with @emdupre's comment) it would be great to take each of the `pytest` commands below and simply add them beneath each of the descriptive sentences above. For example:

> If you know a file will test your change, you can run only that test with:
> 
> ```
> pytest -k my_function $TEDANADIR/tedana/tests/test_file.py
> ```

and so on.